src/mavm.rs

```diff
@@ -977,27 +980,42 @@ pub enum AVMOpcode {
 }
 
 impl Opcode {
-    pub fn is_pure(&self) -> bool {
+    pub fn is_view(&self, _: &TypeTree) -> bool {
+        match self {
+            Opcode::AVMOpcode(AVMOpcode::Inbox)
+            | Opcode::AVMOpcode(AVMOpcode::InboxPeek)
+            | Opcode::AVMOpcode(AVMOpcode::Send)        // 2 sends can't be reordered even though no state is read
+            | Opcode::AVMOpcode(AVMOpcode::Rpush)
+            | Opcode::AVMOpcode(AVMOpcode::PushInsn)
+            | Opcode::AVMOpcode(AVMOpcode::PushInsnImm)
+            | Opcode::AVMOpcode(AVMOpcode::ErrCodePoint)
+            | Opcode::AVMOpcode(AVMOpcode::ErrPush)
+            | Opcode::AVMOpcode(AVMOpcode::PushGas)
+            | Opcode::AVMOpcode(AVMOpcode::Sideload)    // this is special-cased, so we can't reorder these
```

Comment:
I don't think the reordering comment is relevant here.  (It is relevant below, where Sideload is listed as write.)

Reply:
Yeah, it's not. I've removed it